src/internal/utils.js

```diff
@@ -11,7 +11,11 @@ export const kFalse = konst(false)
 export const noop = () => {}
 export const ident = v => v
 
-export const isDev = process.env.NODE_ENV === 'development'
+const hasEnv = typeof process !== 'undefined' && typeof process.env !== 'undefined'
+
+export const env = (key) => hasEnv ? process.env[key] : null
+
+export const isDev = env('NODE_ENV') ? env('NODE_ENV') === 'development' : false
```

Comment:
There's a chance that this won't work with this Webpack configuration:
```js
new DefinePlugin({
  'process.env': JSON.stringify({NODE_ENV: 'development'}),
})
```
Because Webpack only replaces `process.env` and not `process`. This is how it's done in e.g. `create-react-app`: https://github.com/facebookincubator/create-react-app/blob/v0.9.0/packages/react-scripts/config/env.js

I may be mistaken, but I remember I've ran into such issue before.